### PR TITLE
Removes a disgusting hack

### PR DESCRIPTION
**Pull request by SamHPurp**
No longer at round end, will the server have to scrape through all clowns attack logs and increment... now it'll increment when the clown's get hit.

This theoretically will speed up the round-end screen / reduce the hitch that happens.

No CL as this'll effect nothing really player-side.

### CODE REVIEW

**Comment by FlattestGuitar** [comment]
I actually nerfed clown abuse when I added the log collapse system, didn't tell anyone.

**Comment by SamHPurp** [comment]
@FlattestGuitar that was part of why I checked out the code, people were remarking that the numbers had dropped massively months back.

**Comment by FlattestGuitar** [comment]
😇

**Comment by Fox-McCloud** [comment]
A good start, but still no there yet.

You're going to have to check the suit's wearer and if they're truly a clown; you're also doing to want to check for if the attack actually did damage.

**Comment by SamHPurp** [comment]
@Fox-McCloud I can easily add this, and infact decided it may be better without it. I'd say anyone wearing a clown jumpsuit should count as a clown for abuse purposes. You'd want the HoP that was forcefully demoted to clown mid-shift to count for abuse, no?

Also ganging up on them and pushing them and being a jerk (also triggering the hit_reaction()) should count for abuse purposes.

Do you not agree?

**Comment by Lady-Luck** [comment]
Please don't remove @FlattestGuitar he is my friend.

**Comment by tigercat2000** [comment]
@SamHPurp you pinged the wrong person, it's @Krausus

**Comment by Krausus** [comment]
> feel it misses the sort of fringe cases

It would, but whether that's a good thing or a bad thing depends on whether we want "clown abuse" to include everyone dressed as a clown, or just the "true" clowns. I'd say it should count everyone, because abusing someone merely dressed as a clown is still clown abuse, at least in spirit... and, really, isn't it the *spirit* of clown abuse that matters?

Plus it's a joke stat, so, no need to put too much thought into it.

**Comment by taukausanake** [comment]
So if I kidnap someone, put them in a clown suit, and start beating them, healing them, wash rinse repeat, I can meta that endgame score? Or better yet why not have an adminbus command like Anime Mode but instead make it Clown mode and see how high that number can get?

**Comment by SamHPurp** [comment]
> and, really, isn't it the spirit of clown abuse that matters?

Thats exactly my point.

@taukausanake its a joke stat for fun. If you really wish to do that, god speed.

**Comment by ZomgPonies** [comment]
Reminds me of the need to run an event where I create a shelter for abused clowns.

**Comment by taukausanake** [comment]
Wondering now. Would it be possible to put this into human attack code or when a human is attacked? If their job role is clown increment there? Seems more sensible to do it there rather than in the clothing

**Comment by SamHPurp** [comment]
@taukausanake I disagree for the following reasons:

- Its the spirit of clown abuse that matters. If the HoP is a dick and gets demoted to clown, jumpsuit and all, then abuse of him should count for this figure. If you're checking mind this'll only do the roundstart clown.

- For a niche things, adding this check to basically every attack ever, seems entirely wasteful.

**Comment by taukausanake** [comment]
> gets demoted to clown

Then he is by all means a clown and it should count towards the score. I don't feel it's within any spirit to stuff someone in a clown suit and call them a clown. That's kind of the opposite if you ask me.
Would there be a performance issue making such a check at every attack? At that matter, would it count for dead/SSD clowns being attacked or damaged by any means? Those are the only issues I can think of

**Comment by SamHPurp** [comment]
@taukausanake getting demoted to clown doesn't make your `mind` a clown

**Comment by Fox-McCloud** [comment]
>I can simply make your requested change, but feel it misses the sort of fringe cases that @krasus has been arguing to the death over on another PR. Do you not agree?

The key difference is that the metric being tracked there (experience for playtime in a certain job) necessitates that the system be dynamic like that. 

Traditionally, mind role is hard coded and shouldn't be changed--there's a lot of hardcoded checks that make similar assumptions. There's a number of reasons for this; the chief of them being that players can and will abuse changing it to their own advantage, which we don't want; for example, you could join as an assistant, purchase scientist traitor gear, then get another job to something else and buy their unique traitor gear.

It also becomes optimal to set people to Chaplain to get special protections during certain round types, as well.

For consistencies sake (and the fact the system isn't really designed for it), that's why I feel for non-administrative things, it would probably be best to have it hard check the clown role.

File: code/modules/clothing/under/jobs/civilian.dm [hunk 1]
```diff
@@ -72,6 +72,7 @@
 
 /obj/item/clothing/under/rank/clown/hit_reaction()
 	playsound(loc, 'sound/items/bikehorn.ogg', 50, 1, -1)
+	score_clownabuse++
```
Comment (Fox-McCloud) [inline code/modules/clothing/under/jobs/civilian.dm]:
```
if(ishuman(loc))
	var/mob/living/carbon/human/H = loc
	if(H.mind && H.mind.assigned_role == "Clown")
 	 	score_clownabuse++
```

**Comment by Fox-McCloud** [comment]
Please update.

**Comment by SamHPurp** [comment]
@Fox-McCloud Away from my PC... Going to try to use the web-editor. Hold onto your butts.

**Comment by Crazylemon64** [comment]
It's still broken @SamHPurp

**Comment by SamHPurp** [comment]
@Crazylemon64 god damn web-editor... hang on.

**Pull request merged**